File: go/1688.go
func numberOfMatches(n int) int {
    matches := 0
    for n > 1 {
        if n % 2 == 0 {
            n /= 2
            matches += n
        } else {
            n = ((n-1) / 2) + 1
            matches += n - 1
        }
    }
    return matches
}

File: py/1688.py
class Solution:
    def numberOfMatches(self, n: int) -> int:
        matches = 0 
        while n > 1:
            if n % 2 == 0:
                n //= 2
                matches += n
            else:
                n = int((n-1) / 2) + 1
                matches += n-1
        return matches
